### ISSUE TITLE
`UnaryOperation` expression type missing from type definitions

### ISSUE THREAD

**Issue by naddison36** [federicobond/solidity-parser-antlr#86]
See line 104 of IDEX's Exchange contract verified on Etherscan https://etherscan.io/address/0x2a0c0DBEcC7E4D658f48E01e3fA353F44050c208#code
```js
if (!Token(token).transferFrom(msg.sender, this, amount)) throw;
```
This has an expression type of `UnaryOperation`

**Comment by yxliang01** [comment]
Yes. This is confirmed. Example `UnaryOperation` formation is at https://github.com/federicobond/solidity-parser-antlr/blob/918b74290b783f3aae7ba690c7163427e70df51c/test/ast.js#L699 .
